Util/exact_riemann/riemann_rarefaction.H

```diff
@@ -0,0 +1,432 @@
+#ifndef RIEMANN_RAREFACTION_H
+#define RIEMANN_RAREFACTION_H
+
+#include <utility>
+
+#include <AMReX_REAL.H>
+
+#include <eos_type.H>
+#include <eos.H>
+
+
+AMREX_INLINE
+void
+riemann_invariant_rhs(const amrex::Real p, const amrex::Real tau, const amrex::Real u,
+                      const amrex::Real* xn, const int iwave,
+                      amrex::Real& T,
+                      amrex::Real& dtaudp, amrex::Real& dudp) {
+
+    amrex::ignore_unused(u);
+
+    // here, p is out independent variable, and tau, u are the
+    // dependent variables.  We return the derivatives of these
+    // wrt p for integration.
+
+    // T should be an initial guess coming in
+
+    // get the thermodynamics
+
```

Comment:
From the equation of state, we hold the density $\rho = 1/\tau$, the pressure $p$, the composition $X_i$, and the temperature $T$  in `eos_rep_t`. Then we perform all the eos calculations based on $\rho(\tau)$ and $p$, override the input temperature, and compute the sound speed $C$.